Sources/SpyableMacro/Extensions/TypeSyntax+Extensions.swift

```diff
@@ -0,0 +1,263 @@
+import SwiftSyntax
+
+extension TypeSyntax {
+  /// An array of all of the `TypeSyntax`s that are used to compose this object.
+  ///
+  /// Ex: If this is a tuple - (A, B). This will return the two `TypeSyntax`s, A & B.
+  private var nestedTypeSyntaxes: [Self] {
+    for typeSyntaxExposingConformer in typeSyntaxExposingConformers {
+      guard let cast = self.as(typeSyntaxExposingConformer.self) else { continue }
+      return cast.nestedTypeSyntaxes
+    }
+    // Return an empty array for unsupported types
+    return []
```

Comment:
Instead of doing this, we could throw an error, but this would result in error propagation that could ends at `forceCastType` in `FunctionDeclSyntax+Extension.swift` by having that return nil. If we wanted to take it even further, we could continue propagating all the way to the top of the Spy factory and emit a diagnostic as well. 

I'm just not sure if this is worth the trouble and adding `throws` everywhere, as I think the only situation that'd reach this code path is if this computed var is triggered on a `MissingTypeSyntax`, and in that case, I think returning an empty array would probably be fine.

The same applies to `erasingGenericTypes()` below.